src/comm.jl

```diff
@@ -3,29 +3,38 @@
 
 An MPI Communicator object.
 """
-@mpi_handle Comm MPI_Comm
+mutable struct Comm
+    val::MPI_Comm
+end
+Base.:(==)(a::Comm, b::Comm) = a.val == b.val
+Base.cconvert(::Type{MPI_Comm}, comm::Comm) = comm.val
```

Comment:
This should be done for all handles
```suggestion
Base.cconvert(::Type{MPI_Comm}, comm::Comm) = comm
Base.unsafe_convert(::Type{MPI_Comm}, comm::Comm) = comm.val
```

Reply:
Done for all handles.